OpenSim/Analyses/StaticOptimizationTarget.cpp

```diff
@@ -479,7 +479,8 @@ objectiveFunc(const Vector &parameters, const bool new_parameters, Real &perform
     int na = _model->getActuators().getSize();
     double p = 0.0;
     for(int i=0;i<na;i++) {
-        p +=  pow(fabs(parameters[i]),_activationExponent);
+        if (ScalarActuator* act = dynamic_cast<ScalarActuator*>(&_model->getActuators().get(i)))
```

Comment:
That may be true since we introduced  the capability of having non scalar actuators. What would you propose? Unwind that capability?

In any case, if an Analysis/Controller, etc needs to operate on a certain type of Actuator it should generate that list prior to simulation/computation time. It makes little sense to filter by type again and again.